duck/call.py

```diff
@@ -0,0 +1,171 @@
+class Call(object):
+    """A Call object.
+
+    A representation of a call made against a ``Mock`` object. It is
+    similar to :class:`unittest.mock._Call`, but is designed to be more
+    easily reasoned about.
+
+    Args:
+        args (list): Positional arguments provided to the call.
+        kwargs (dict): Keyword arguments provided to the call.
+    """
+    def __init__(self, *args, **kwargs):
+        self.args = []
+        if args:
+            self.args = args
+
+        self.kwargs = {}
+        if kwargs:
+            self.kwargs = kwargs
+
+    def has_args(self, *args, **kwargs):
+        """Returns ``True`` if the call was made with a superset of the provided
+        arguments and keyword arguments, and ``False`` otherwise.
+
+        If provided, positional arguments must start at ``0`` and be in order.
+
+        Comparisons are performed with ``==``, not ``is``.
+        """
+        results = []
+
+        # If the given arguments exceed the number of actual arguments, this
+        # cannot be true, and we can stop here.
+        if len(args) > len(self.args):
+            return False
+
+        # Determine if the given argument(s) are present in the Call.
+        args_results = []
+        for idx, actual in enumerate(args):
+            if actual == self.args[idx]:
+                args_results.append(True)
+            else:
+                args_results.append(False)
+
+        results.append(all(args_results))
+
+        # If the given keyword arguments exceed the number of actual keyword
```

Comment:
It short circuits any further processing if you send more keyword arguments than there actually are.  Granted, it would be caught by the iteration below, but why do that work if you've asked for something clearly incorrect?